### ISSUE TITLE
Building for production

### ISSUE THREAD

**Issue by AasmundN** [userquin/vuetify-nuxt-module#74]
Having an issue when trying to build the app for production: https://ibb.co/HrsK1F3.

Here is my nuxt config:

```ts
export default defineNuxtConfig({
   css: ["@/assets/scss/index.scss"],

   // add modules
   modules: ["@nuxtjs/i18n",  "vuetify-nuxt-module"],

   i18n: {
      locales: [
         { code: "en", iso: "en-US", file: "en.ts" },
         { code: "es", iso: "es-ES", file: "es.ts" },
      ],

      lazy: true,
      langDir: "lang",
      defaultLocale: "en",
      detectBrowserLanguage: false,
      strategy: "no_prefix",
      baseUrl: "https://deploii.no/",

      vueI18n: "./config/i18n.config.ts",
   },

   vuetify: {
      moduleOptions: {
         styles: {
            configFile: "assets/scss/_variables.scss",
         },
      },
      vuetifyOptions: "./config/vuetify/vuetify.config.ts",
   },
})

**Comment by bogdanciuca** [comment]
@AasmundN you need to add `vuetify` as a dependency.

**Comment by userquin** [comment]
@AasmundN  Try disabling `inlineSSRStyles` in your Nuxt config file, I'll try to figure out what's happening, the paths for scss files using absolute paths, should be prefixed with `/@fs/`, Vuetify needs a deep review (disabling inline SSR Styles, the error should go away):
```ts
experimental: {
    inlineSSRStyles: false,
}
```

![imagen](https://github.com/userquin/vuetify-nuxt-module/assets/6311119/a2cef2db-1136-470d-892b-0d7afbc13f4a)

**Comment by AasmundN** [comment]
> @AasmundN you need to add `vuetify` as a dependency.

No? `vuetify-nuxt-module` adds it as one of its dependencies.

@userquin Disabling `inlineSSRStyles` seems to have fixed the problem. I'm guessing doing that shouldn't be necessary?

**Comment by userquin** [comment]
> @userquin Disabling `inlineSSRStyles` seems to have fixed the problem. I'm guessing doing that shouldn't be necessary?

Experimental `inlineSSRStyles` is enabled by default, we need to change the custom Vuetify styles plugin here to resolve/load SSR styles properly, I'm trying to fix it.

**Comment by userquin** [comment]
It seems using Vuetify `configFile` styles and Nuxt `inlineSSRStyles` are imcompatible, there is no way to get individual chunks

**Comment by AasmundN** [comment]
Okay. Is that a problem? I'm not really familiar with what _inlineSSRStyles_ does. Or how the styles are handled by Nuxt and Vite for that matter.

**Comment by userquin** [comment]
The problem is the nuxt nitro renderer, since we've some .sass references we will receive the request, but it is missing, I have it working just returning an empty string when receiving the id in Vite load hook with SSR flag: we'll receive requests like this `/plugin-vuetify/lib/components/VDialog/VDialog.sass?inline&used`.

Since we're using Vuetify styles via `configFile` we'll get the styles in the `vuetify-theme-stylesheet`, maybe I can fix the error, but I have no idea if it has side effects...

**Comment by userquin** [comment]
This is the page without inlining SSR styles:

![imagen](https://github.com/userquin/vuetify-nuxt-module/assets/6311119/dee808ba-38f3-4745-9348-fd822c25f223)

Inlining SSR styles:

![imagen](https://github.com/userquin/vuetify-nuxt-module/assets/6311119/26851e5c-fc97-4484-93ac-2ee8b43da935)

**Comment by userquin** [comment]
Vuetify should avoid using SASS...

**Comment by userquin** [comment]
It is not working...

**Comment by userquin** [comment]
@AasmundN You can try it adding this simple Vite plugin to your nuxt config file, removing `inlineSSRStyles: false` from experimental option:
```ts
vite: {
  plugins: [{
    name: 'vuetify-nuxt-inline-ssr-styles',
    enforce: 'pre',
    load(id, options) {
      if (options?.ssr && id.startsWith('/plugin-vuetify/lib/'))
        return ''
    }
  }]
}

**Comment by AasmundN** [comment]
Adding the Vite plugin also seems to work, and it does reduce the size of the css-file like you showed. However when I load the page from the production build there is a little "flick" of the styles, as if the styles are being applied a little after the html appears.

**Comment by AasmundN** [comment]
Yeah, the vite plugin doesn't seem to be working properly. Is using inlineSSRStyles important?

**Comment by AasmundN** [comment]
Still in the process of deciding what framework to choose for styling in my Nuxt project, and I would really like to use Vuetify as I have used it in other SPA, Vite projects. However, adding Vuetify to Nuxt has been quite painful and things don't seem to work properly at every other turn. I'm not so familiar with how Vite works, but I've been trying to understand wether or not this inline styles thing is a deal breaker or not, and thus if I should choose some other styling framework instead. Or would it just marginally change the load time. If it is important, will eventually work more into fixing it? Thank you for your work, a module like this one is really needed.

**Comment by AndreyYolkin** [comment]
Fixed in the v1.0.0-beta line.

This was the old custom styles plugin interacting with `experimental.inlineSSRStyles` (the workaround at the time was disabling inline SSR styles). The module has since migrated to `@vuetify/unplugin-styles` and that `inlineStyles` override was removed.

Verified: a production build (`nuxi build`) of an SSR app with `vuetify.moduleOptions.styles.configFile` (SCSS) completes successfully, and the `configFile` + `build` path is covered by `test/e2e/styles-ssr.spec.ts`. Please upgrade to the latest beta and reopen if it recurs.
